aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -293,19 +293,36 @@ pub struct AggregatorData {
     destroyed_aggregators: BTreeSet<AggregatorID>,
     // All aggregator instances that exist in the current transaction.
     aggregators: BTreeMap<AggregatorID, Aggregator>,
+    // If aggregators are not enabled, then all aggregators will directly store the Data instead of deltas.
+    is_aggregator_enabled: bool,
 }
 
 impl AggregatorData {
+    /// Initializes AggregatorData with default values
+    pub fn new(is_aggregator_enabled: bool) -> Self {
+        Self {
+            new_aggregators: Default::default(),
+            destroyed_aggregators: Default::default(),
+            aggregators: Default::default(),
+            is_aggregator_enabled,
+        }
+    }
+
     /// Returns a mutable reference to an aggregator with `id` and a `limit`.
     /// If transaction that is currently executing did not initilize it), a new
     /// aggregator instance is created, with a zero-initialized value and in a
     /// delta state.
     /// Note: when we say "aggregator instance" here we refer to Rust struct and
     /// not to the Move aggregator.
     pub fn get_aggregator(&mut self, id: AggregatorID, limit: u128) -> &mut Aggregator {
+        let state = if self.is_aggregator_enabled {
```

Comment:
This can be dangerous because the callers of `get_aggregator` (native functions in this case) has to ensure the value is read for it to be `Data`.

Reply:
Actually, it might not even work: if you call `read_and_materialize` after you called `get_aggregator` the read will see it is `Data` and return a value of 0 :(

Reply:
I see. If `is_aggregator_enabled` flag is set to `false`, I can read the aggregator value from the table, and initialize with that value instead of zero. Would that work?

Reply:
Yes! I still would move the flag to context, it doesn't belong to aggregator data in my opinion. Maybe we can change  the signature of `get_aggregator` to take a flag `materialize` or something?

Reply:
I moved the flag to context, and initialized the aggregator by reading value from the table. Please let me know what you think.